core/src/main/scala/filodb.core/downsample/DownsampledTimeSeriesShard.scala

```diff
@@ -139,47 +140,51 @@ class DownsampledTimeSeriesShard(rawDatasetRef: DatasetRef,
 
   private def hour(millis: Long = System.currentTimeMillis()) = millis / 1000 / 60 / 60
 
-  def recoverIndex(): Future[Unit] = {
+  def recoverIndex(): Future[Long] = {
     if (downsampleConfig.enablePersistentIndexing) {
       partKeyIndex.getCurrentIndexState() match {
         case (IndexState.Empty, _)              =>
           logger.info("Found index state empty, bootstrapping downsample index")
-          recoverIndexInternal
+          recoverIndexInternal(None)
+        case (IndexState.Synced, checkpointMillis)             =>
+          logger.info("Found index state empty, bootstrapping downsample index")
+          recoverIndexInternal(checkpointMillis)
         case _                                  =>
           // FIXME: Part ids in case of indexRefresh are assumed to monotonically increase, however, in case recovery
           //  is done from an existing timestamp, we need a way not only to store the state of the ids recycled but
           //  latest allocated partId to avoid assigning the same id again.
-          logger.info(s"Nothing to recover the index for dataset=$indexDataset shard=$shardNum" +
-                      s" starting index refresh thread")
-          indexRefresh().runToFuture(housekeepingSched)
+          //logger.info(s"Nothing to recover the index for dataset=$indexDataset shard=$shardNum" +
+          //            s" starting index refresh thread")
+          //indexRefresh().runToFuture(housekeepingSched)
```

Comment:
Why comment out code? This is needed to start the index refresh thread

Reply:
log but don't remove it